TACC/Core-Portal
# task/FP-304: Separate custom settings from secrets

## Overview: ##

Custom django settings that are not secrets, i.e. configurations, do not need to be hidden, are configurable, and can be tracked. We can use a composition of `settings.py` files to provide all the settings we need.

## Related Jira tickets: ##

* [FP-304](https://jira.tacc.utexas.edu/browse/FP-304)

## Summary of Changes: ##

New files:

- `settings_custom.py`: Custom settings/configuration for a given portal, excluding any sensitive secrets or authentication. These will be provided for a portal via the [Camino](https://github.com/TACC/Camino)/[Core-Portal-Deployments](https://github.com/TACC/Core-Portal-Deployments) workflow launched from [Jenkins](https://jenkins01.tacc.utexas.edu/view/WMA%20CEP/job/Core_Portal_Deploy/).
- `settings_default.py`: Default configurations to be used during local development
- `settings_overrides.py`: An optional file to override loaded settings from either `settings_custom.py` or `settings_default.py`. For example, this can be used during local development to make changes to configurations without needing to create a git diff.

Order of operations for loading these new files:
- Loads a `settings_custom.py` file if one exists, otherwise loads `settings_default.py` for local development.
- Loads a `settings_overrides.py` file if one exists

## Testing Steps: ##
1. Paste the "NOTE" portion of the [Stache](https://stache.utexas.edu/entry/bedc97190d3a907cb44488785440595c) Entry for `Core Portal Settings Secret` into `settings_secret.py`
2. Create the file `server/portal/settings/settings_overrides.py` and paste your ngrok url
3. Run the portal and ensure everything functions

## Notes
- [This PR ](https://github.com/TACC/Core-Portal-Deployments/pull/1)in Core-Portal-Deployments is the relevant PR to implement this in the deployment workflow.
